addons/mail/static/src/core/common/message_model.js

```diff
@@ -503,17 +508,24 @@ export class Message extends Record {
         );
     }
 
-    async remove() {
-        await rpc("/mail/message/update_content", {
+    async remove(removeFromThread = false) {
+        const data = await rpc("/mail/message/update_content", this.removeParams);
+        this.store.insert(data, { html: true });
+        if (removeFromThread) {
+            this.thread.messages = this.thread.messages.filter((message) => message.id !== this.id);
+        }
```

Comment:
thread is not necessarily existing/known on message, we usually guard it. (probably not an issue in this flow but better be safe)